File: toy/react-spring/src/main/java/me/kzv/reactspring/todo/TodoController.java
package me.kzv.reactspring.todo;

import lombok.RequiredArgsConstructor;
import lombok.extern.log4j.Log4j2;
import me.kzv.reactspring.common.dto.PageRequest;
import me.kzv.reactspring.common.dto.PageResponse;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RequiredArgsConstructor
@RestController
@Log4j2
@RequestMapping("/api/todo")
public class TodoController {
    private final TodoService todoService;

    @GetMapping("/{tno}")
    public TodoDto get(@PathVariable Long tno) {
        return todoService.get(tno);
    }

    @GetMapping("/list")
    public PageResponse<TodoDto> list(PageRequest request) {
        log.info(request);
        return todoService.list(request);
    }

    @PostMapping("/")
    public Map<String, Long> register(@RequestBody TodoDto dto) {
        log.info("TodoDTO: " + dto);
        var id = todoService.register(dto);

        return Map.of("id", id);
    }

    @PutMapping("/{tno}")
    public Map<String, String> modify(@PathVariable Long tno, @RequestBody TodoDto dto) {
        dto.setId(tno);
        todoService.modify(dto);
        return Map.of("result", "SUCCESS");
    }

    @DeleteMapping("/{tno}")
    public Map<String, String> remove(@PathVariable Long tno) {
        todoService.remove(tno);

        return Map.of("result", "SUCCESS");
    }
}

File: hello-kotlin/todo/src/main/kotlin/me/kzv/todo/controller/TodoController.kt
package me.kzv.todo.controller

import me.kzv.todo.controller.dto.TodoListResponseDto
import me.kzv.todo.controller.dto.TodoRequestDto
import me.kzv.todo.controller.dto.TodoResponseDto
import me.kzv.todo.service.TodoService
import org.springframework.http.ResponseEntity
import org.springframework.http.ResponseEntity.noContent
import org.springframework.http.ResponseEntity.ok
import org.springframework.web.bind.annotation.*

@RestController
class TodoController(
    private val todoService: TodoService
) {
    @GetMapping
    fun getAll() = ok(TodoListResponseDto.of(todoService.findAll())) // ResponseEntity.ok

    @GetMapping("/{id}")
    fun get(@PathVariable id: Long) = ok(todoService.findById(id))

    @PostMapping
    fun create(@RequestBody requestDto: TodoRequestDto) = ok(TodoResponseDto.of(todoService.create(requestDto)))

    @PostMapping("/{id}")
    fun update(@PathVariable id: Long, @RequestBody requestDto: TodoRequestDto) =
        ok(TodoResponseDto.of(todoService.update(id, requestDto)))

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Unit> {
        todoService.delete(id)
        return noContent().build()
    }
}